onnxruntime/core/platform/windows/logging/etw_sink.h

```diff
@@ -41,6 +45,66 @@ class EtwSink : public ISink {
   // EtwTracingManager to ensure we cleanly unregister it
   static std::atomic_flag have_instance_;
 };
+
+class EtwRegistrationManager {
+ public:
+  using EtwInternalCallback = std::function<void(LPCGUID SourceId, ULONG IsEnabled, UCHAR Level,
+                                                 ULONGLONG MatchAnyKeyword, ULONGLONG MatchAllKeyword,
+                                                 PEVENT_FILTER_DESCRIPTOR FilterData, PVOID CallbackContext)>;
+
+  // Singleton instance access
+  static EtwRegistrationManager& Instance();
+
+  // Check if ETW logging is enabled
+  bool IsEnabled() const;
+
+  // Get the current logging level
+  UCHAR Level() const;
+
+  Severity MapLevelToSeverity();
+
+  // Get the current keyword
+  uint64_t Keyword() const;
+
+  // Get the ETW registration status
+  HRESULT Status() const;
+
+  void RegisterInternalCallback(const EtwInternalCallback& callback);
+
+ private:
+  EtwRegistrationManager();
+  ~EtwRegistrationManager();
+  void LazyInitialize();
+
+  // Copy and move constructors/operators are disabled
+  EtwRegistrationManager(const EtwRegistrationManager&) = delete;
+  EtwRegistrationManager& operator=(const EtwRegistrationManager&) = delete;
+  EtwRegistrationManager(EtwRegistrationManager&&) = delete;
+  EtwRegistrationManager& operator=(EtwRegistrationManager&&) = delete;
```

Comment:
nit:  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE

Reply:
Thx. Didn't know this ORT macro existed. Will use it instead @skottmckay